Sync shell history

kachick opened the pull request:
Resolves #173


This is a way, but... 🤔

kachick committed:
Integrate atuin without server sync

kachick committed:
Disable hijacking up-arrow by atuin

kachick commented:
https://github.com/atuinsh/atuin/issues/816 👀

ellie commented on home-manager/home.nix:
```diff
@@ -150,4 +150,18 @@
+    ];
+
+    settings = {
+      sync_address = "https://127.0.0.1";
```
you probably also want `auto_sync = false` 😁

kachick replied:
Thanks for letting me know!

I may use atuin with self hosting 🐋

kachick commented on home-manager/bash.nix:
```diff
@@ -6,6 +6,7 @@
   programs.zoxide.enableBashIntegration = true;
   programs.fzf.enableBashIntegration = true;
   programs.rtx.enableBashIntegration = true;
+  programs.atuin.enableBashIntegration = true;
```
https://github.com/atuinsh/atuin/discussions/1683

hmm... I want to keep bash as plain as possible, it is the first intention of preferreing zsh https://github.com/kachick/times_kachick/issues/184

Want sytaxhighlighting, bash cannot it, ble.sh realize it, ble.sh looks so big project for me

kachick committed:
Merge branch 'main' into secret-history

kachick committed:
Clarify testing the atuin ctrl+r

kachick committed:
Run atuin server on algae

kachick committed:
Merge branch 'main' into secret-history

kachick committed:
Disable atuin on bash

However brush recommends ble, I may reconsider it in future

kachick committed:
Disable atuin ctrl-r

kachick committed:
Adjust atuin server config

kachick committed:
Merge branch 'main' into secret-history

kachick commented on home-manager/common.nix:
```diff
@@ -210,4 +210,21 @@
+
+    settings = {
+      # auto_sync = true; # TODO: Consider enabling after test
+      sync_address = "https://algae";
```
https://github.com/kachick/dotfiles/pull/1265 might be better than tailscale

kachick replied:
```suggestion
      sync_address = "https://algae.local";
```

However the mDNS is flaky

kachick committed:
Merge branch 'main' into secret-history

kachick committed:
atuin: openRegistration

kachick committed:
Clarify port for self hosted atuin

kachick committed:
Test with http

kachick committed:
Enable SSH portforwarding for atuin

kachick committed:
Update config around atuin

kachick committed:
Use caddy for atuin server

kachick committed:
Update config and comments about atuin

[ci skip]

kachick commented on nixos/hosts/algae/default.nix:
```diff
@@ -68,10 +69,39 @@
+  services.caddy = {
+    enable = true;
+
+    acmeCA = "https://acme-v02.api.letsencrypt.org/directory";
```
tailscale should be easier if I should introduce external service for HTTPS

kachick committed:
Merge branch 'main' into secret-history

kachick committed:
Comment-out unused atuin code

kachick committed:
Run tailscale serve on systemd

kachick committed:
Set user

kachick committed:
Ensure to avoid atuin default API

kachick commented on nixos/configuration.nix:
```diff
@@ -249,4 +249,12 @@
   # https://github.com/NixOS/nixpkgs/blob/nixos-25.05/nixos/modules/security/sudo-rs.nix
   security.sudo-rs.enable = true;
+
+  networking.hosts = {
```
However this mapping does not work atuin clients on home-manager

kachick commented on nixos/hosts/algae/default.nix:
```diff
@@ -68,10 +69,39 @@
+    # openFirewall = true; # Unnecessary to enable for SSH portforwarding
+    # default 127.0.0.1 does not accept accesses from other hosts. However it is okay for current SSH portforwarding use-case
+    # host = "127.0.0.1";
+    openRegistration = true;
```
What is the difference of user and e-mail?

kachick replied:
I'm not sure why e-mail is required in this context

selfup-runner[bot] committed:
Bump flake.lock and related dependencies (#1320)

* flake.lock: Update

Flake lock file updates:

• Updated input 'edge-nixpkgs':
    'https://releases.nixos.org/nixos/unstable/nixos-25.11pre880095.5e2a59a5b1a8/nixexprs.tar.xz?narHash=sha256-u0JUo46QSoXnjLaezAM75wRNuxVMVbm5OxHH122TeTY%3D' (2025-10-20)
  → 'https://releases.nixos.org/nixos/unstable/nixos-25.11pre883951.6a08e6bb4e46/nixexprs.tar.xz?narHash=sha256-5bRIcZ2GZ%2BWo3XrFVcFWGDIjycGp53B/9TNj1r4Iwsc%3D' (2025-10-26)
• Updated input 'nixos-wsl':
    'github:nix-community/NixOS-WSL/7014f72d94af88a6db845231a49f8566cbc70c3e?narHash=sha256-Ntp1XZdMpIykkAvYTOjX2jnO86RQ2YOjqVH4cD%2BHe24%3D' (2025-10-15)
  → 'github:nix-community/NixOS-WSL/25a1160f47e16c017cee93acf58f3649ad87c213?narHash=sha256-qJ2TP6WIT7YUL4oZSEvib5j19df4V55WS79OVGmX72s%3D' (2025-10-26)
• Updated input 'nixpkgs':
    'https://releases.nixos.org/nixos/25.05/nixos-25.05.811461.81b927b14b7b/nixexprs.tar.xz?narHash=sha256-H/cHnBjaol3hAwJwnShW48fIWbJeROctqovDHDlzcQw%3D' (2025-10-19)
  → 'https://releases.nixos.org/nixos/25.05/nixos-25.05.811621.c8aa8cc00a5c/nixexprs.tar.xz?narHash=sha256-/UR943qCkSDpSJrABtcHn56cglxJoX45ezW1LtEHpUw%3D' (2025-10-24)
• Updated input 'nixpkgs-darwin':
    'https://releases.nixos.org/nixpkgs/25.05-darwin/nixpkgs-darwin-25.05pre811497.33c6dca0c0cb/nixexprs.tar.xz?narHash=sha256-M8ivoTcUHC4q7fHubxKQCElzwM1qI1xy11v4NBSeMKM%3D' (1980-01-01)
  → 'https://releases.nixos.org/nixpkgs/25.05-darwin/nixpkgs-darwin-25.05pre811753.08478b816182/nixexprs.tar.xz?narHash=sha256-%2Btt3r2tb0a6yce59V48pQPn8exKGDu%2Br2WVPfN7ySsM%3D' (1980-01-01)

* Update go.mod and go.sum

---------

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump actions/upload-artifact from 4 to 5 (#1322)

Bumps [actions/upload-artifact](https://github.com/actions/upload-artifact) from 4 to 5.
- [Release notes](https://github.com/actions/upload-artifact/releases)
- [Commits](https://github.com/actions/upload-artifact/compare/v4...v5)

---
updated-dependencies:
- dependency-name: actions/upload-artifact
  dependency-version: '5'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

kachick committed:
Prevent access to external service with the customizing config

kachick committed:
Enable ctrl+r on atuin

kachick committed:
Update legacy comments

kachick committed:
Merge branch 'main' into secret-history-local

kachick committed:
Share atuin port except home-manager

kachick committed:
Prefer fzf again

kachick committed:
Correct env in comment

kachick committed:
Relax systemd retry

kachick committed:
Prefer nobody account for this role

kachick commented on home-manager/common.nix:
```diff
@@ -220,6 +220,31 @@
+      "--disable-up-arrow"
+
+      # NOTE: Add another keybind if I found suitable one
+      "--disable-ctrl-r" # Prefer fzf key-bindings. Use `atuin search --interactive` instead.
```
>enter	Execute selected item
>ctrl + f / ➡️	Move the cursor to the right

From the doc, the behavior is incorrect: https://docs.atuin.sh/configuration/key-binding/#atuin-ui-shortcuts

kachick committed:
Clarify enter_accept

kachick committed:
Add comment about bash integrations

kachick committed:
Merge branch 'main' into secret-history

kachick commented:
:thinking: While using this revison of dotfiles, the algae network is much flaky...

kachick committed:
yaneuraou: 9.01-unstable -> 9.01

Diff: https://github.com/yaneurao/YaneuraOu/compare/33dce0bfa363f63d99977c29b3d6ab40ff896138...v9.01git

kachick committed:
Merge branch 'main' into secret-history

kachick committed:
Revert "Prefer nobody account for this role"

This reverts commit 2c6b2010f536e5d00abc3cf21c11ce83117d067c.

kachick committed:
Reapply "Prefer nobody account for this role"

This reverts commit 070e70aa15819d0f0ec075995e2b16521d58c3b9.

kachick committed:
Merge branch 'main' into secret-history
